### ISSUE TITLE
API: query vulnerabilities by cve id

### ISSUE THREAD

**Issue by huanceng** [google/osv.dev#1016]
The current api `/v1/vulns/{id}` returns a vulnerability object for a given **OSV ID**. If a cve id is given, the api will return `Bug not found.`

For example, GHSA-h592-38cm-4ggp and its corresponding cve id CVE-2017-15095.

Do you plan to develop such an API?

**Comment by oliverchang** [comment]
This does sound useful thanks, tracking as a FR.

**Comment by rdeepc** [comment]
Hi @oliverchang,

I am interested in working on this.

**Comment by andrewpollock** [comment]
More abstractly, it sounds like you're wishing for the `/v1/vulns{id}` endpoint to support lookups by alias as well as the primary ID?

**Comment by huanceng** [comment]
Hi @andrewpollock , `/v1/vulns/{id}` or a new endpoint are both acceptable.

**Comment by andrewpollock** [comment]
Some interim thoughts to improve this user experience:

```
$ GET -s https://api.osv.dev/v1/vulns/CVE-2025-0001
404 Not Found
{"code":5,"message":"Bug not found."}
```

* include a link to https://osv.dev/faq in the message text
* add a new FAQ entry about bugs not found/404s from the `/v1/vulns/{id}` API endpoint
* add to https://google.github.io/osv.dev/post-v1-query/ about querying by first class vulnerabilities versus aliases

**Comment by andrewpollock** [comment]
Related: #760

**Comment by jess-lowe** [comment]
I believe this was closed by #3462. Feel free to reopen if not solved!

**Comment by cmoulliard** [comment]
I don't think that the issue is fixed as we continue to get a HTTP 404 response with body
```
❯ http https://api.osv.dev/v1/vulns/CVE-2026-40970
HTTP/1.1 404 Not Found
Content-Length: 89
alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
content-type: application/json
date: Fri, 03 Jul 2026 12:22:34 GMT
server: Google Frontend
...

{
    "code": 5,
    "message": "Bug not found, but the following aliases were: GHSA-c96x-rpm4-349p"
}
```
If I extract the GHSA id, then I can requery the API to find the record
```
❯ http https://api.osv.dev/v1/vulns/GHSA-c96x-rpm4-349p
...
{
    "affected": [
        {
            "database_specific": {
                "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2026/04/GHSA-c96x-rpm4-349p/GHSA-c96x-rpm4-349p.json"
            },
            "package": {
                "ecosystem": "Maven",
                "name": "org.springframework.boot:spring-boot-elasticsearch",
                "purl": "pkg:maven/org.springframework.boot/spring-boot-elasticsearch"
            },
            "ranges": [
                {
                    "events": [
                        {
                            "introduced": "4.0.0"
                        },
                        {
                            "fixed": "4.0.6"
                        }
                    ],
                    "type": "ECOSYSTEM"
                }
            ],
            "versions": [
                "4.0.0",
                "4.0.1",
                "4.0.2",
                "4.0.3",
                "4.0.4",
                "4.0.5"
            ]
        }
    ],
    "aliases": [
        "CVE-2026-40970"
    ],
...
```

Can we then reopen the ticket please @jess-lowe ?

**Comment by jess-lowe** [comment]
Hey @cmoulliard 

> I don't think that the issue is fixed as we continue to get a HTTP 404 response with body

Can you please clarify what isn't fixed? Is it that you're receiving a 404 response, when it should be another response to suggest you are able to be redirected? Or is it that you'd prefer to be redirected regardless of the ID? 

I believe at the time of implementing this change, the logic was that by querying by an ID, you should receive exactly the record you queried for, but be informed about any aliases. Think of it like ordering a gluten free carrot cake, and us saying "we don't have any gluten free carrot cake in stock, but we do have a vegan carrot cake".  That may matter to you or it may not. 

Hope that makes sense? If you could provide the usecase that would be very helpful to us :)

**Comment by cmoulliard** [comment]
> Can you please clarify what isn't fixed? Is it that you're receiving a 404 response, when it should be another response to suggest you are able to be redirected? Or is it that you'd prefer to be redirected regardless of the ID?

I was simply expected that "osv.dev" API supports to query using the Id or an alias Id as this is needed when we use the CVE id and not the GHSA id as the CVE id is currently the de facto id.
